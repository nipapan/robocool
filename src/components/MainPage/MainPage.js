import React from 'react';

import CardList from "../../components/CardList/CardList";
import SearchBox from '../../components/SearchBox/SearchBox';
import Scroll from "../../components/Scroll/Scroll";
import ErrorBoundary from "../../components/ErrorBoundary/ErrorBoundary";
import Header from "../../components/Header/Header";
import CounterButton from "../../components/CounterButton/CounterButton";

class MainPage extends React.Component {
   componentDidMount() {
      this.props.onRequestRobots();
   }

   filterRobots = () => {
      return this.props.robots.filter((robot) => {
         return robot.name.toLowerCase().includes(this.props.searchText.toLowerCase());
      });
   }

   render() {
      const { onSearchChange, isPending } = this.props;

      return isPending?
         <p>Loading...</p> :
         (
            <div className='tc'>
               <Header />
               <CounterButton />
               <SearchBox onSearchChange={onSearchChange} />
               <Scroll>
                  <ErrorBoundary>
                     <CardList robots={this.filterRobots()} />
                  </ErrorBoundary>
               </Scroll>
            </div>
         );
   }
};

export default MainPage;

