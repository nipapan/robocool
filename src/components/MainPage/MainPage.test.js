import {shallow} from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
   const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchText: '',
      isPending: false
   }
   wrapper = shallow(<MainPage {...mockProps}/>);
})

it('renders Mainpage without crashing', () => {
   expect(wrapper).toMatchSnapshot();
})

it('filters robots correctly (with keyword that match)', () => {
   const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [{
         id: 3,
         name: 'Palo Tom',
         email: 'palotom@gmail.com'
      }],
      searchText: 't',
      isPending: false
   }
   const wrapper2 = shallow(<MainPage {...mockProps2}/>);
   expect(wrapper2.instance().filterRobots()).toEqual(mockProps2.robots);
})

it('filters robots correctly (with keyword that does not match)', () => {
   const mockProps3 = {
      onRequestRobots: jest.fn(),
      robots: [{
         id: 3,
         name: 'Palo Tom',
         email: 'palotom@gmail.com'
      }],
      searchText: 'aom',
      isPending: false
   }
   const expectedResults = [];
   const wrapper3 = shallow(<MainPage {...mockProps3}/>);
   expect(wrapper3.instance().filterRobots()).toEqual(expectedResults);
})

it('renders MainPage correctly when isPending is true', () => {
   const mockProps4 = {
      onRequestRobots: jest.fn(),
      robots: [{
         id: 3,
         name: 'Palo Tom',
         email: 'palotom@gmail.com'
      }],
      searchText: '',
      isPending: true
   }
   const wrapper4 = shallow(<MainPage {...mockProps4}/>);
   expect(wrapper4).toMatchSnapshot();
})
