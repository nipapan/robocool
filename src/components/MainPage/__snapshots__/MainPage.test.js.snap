// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage correctly when isPending is true 1`] = `ShallowWrapper {}`;

exports[`renders Mainpage without crashing 1`] = `ShallowWrapper {}`;
